### ISSUE TITLE
Create pure text export

### ISSUE THREAD

**Issue by wleoncio** [citation-file-format/cff-converter-python#144]
I would really like the converter to have an option to export a CITATION.cff file to an APA-like format. I am far from being a Python expert, but I think I do this myself using one of the classes (RisObject looks simple enough to adapt) as a base.

Using the example from the README file, what I have in mind is for a command like `cffconvert -f apalike` to yield something similar to

```
Spaaks J.H., Klaver T. (2018). cff-converter-python (version 1.0.0). doi: 10.5281/zenodo.1162057
```

I believe this could be very useful for people wanting to cite software in less structured contexts (an e-mail, a pure-text document editor, etc.)

I've already gone ahead and forked the project, but as I started working on it I remembered there might be some contributing guideline somewhere I should be reading first. Then I stopped working and am waiting for a greenlight before I proceed.

Cheers and thanks for starting this useful project. I've heard about it last week during a workshop in Norway.

**Comment by jspaaks** [comment]
Hi Waldir, thanks for this. I agree that this is a useful addition, so I'd say go right ahead!

Be advised that I recently added contracts for tests, basically they are a way of formalizing what tests a test class should implement. The contracts are found in ``tests/contracts``. Each test class should inherit from its corresponding contract (in addition to the standard unittest class).

I recently reorganized the tests such that the ``tests`` directory branches into subdirectories for each supported version of the cff schema (``1.0.3`` and ``1.1.0`` at the moment), and for each of those, there can be an unlimited number of subdirectories, which are simply numbered. The numbered directories are meant to be self-contained units except for the contracts. It should contain a CFF file that is compliant with the corresponding schema version, as well as its converted counterparts in ``bibtex.bib``, ``ris.txt``, ``apalike.txt`` etc.

BTW note that the RIS format has just one test at the moment (#143).

Good luck, I look forward to your pull request. Let me know if I can help you with something.

Best regards, 
Jurriaan

PS I'd be interested to read more about the workshop you mentioned, are there any publicly accessible materials like a website or something? Thanks!

**Comment by wleoncio** [comment]
Thank you for the approval, @jspaaks! I am a bit familiar with unit testing on Python, but contracts are a brand-new thing to me. In any case, I will take a look at it and make sure the new functionality doesn't break any previously-passing tests. I'll also see if I can manage to write some tests for the proposed features before the PR.

So the worshop I mentioned was called ["Social coding and open software"](https://www.ub.uio.no/english/courses-events/events/all-libraries/2020/research-bazaar-2020.html), it was part of the [Research Bazaar 2020](https://www.ub.uio.no/english/courses-events/events/all-libraries/2020/research-bazaar-2020.html) festival organized at the University of Oslo. It was given by @annefou and the slides are available at https://coderefinery.org/lessons/ ([direct link to CFF mention](https://cicero.xyz/v3/remark/0.14.0/github.com/coderefinery/social-coding/master/talk.md/#35))

**Comment by wleoncio** [comment]
Just an update. I've finished implementing the new format, here is what it currently looks like:
![bilde](https://user-images.githubusercontent.com/8234768/75155485-75764580-5710-11ea-861a-73226b5adca8.png)

Up next, I'll work on implementing the tests according to the contracts specifications. The PR should come right after that.

**Comment by jspaaks** [comment]
Very nice! Let me know if you need assistance with the tests.
-Jurriaan

**Comment by wleoncio** [comment]
Hi, I've managed to add passing unit tests for all the 1.1.0 version subfolders, but on "1.0.3" and "unsupported" the tests are yielding setup errors. This is one of them:

```python
________________________________________ ERROR at setup of ApalikeObjectTest.test_author _________________________________________

cls = <class '_pytest.runner.CallInfo'>, func = <function call_runtest_hook.<locals>.<lambda> at 0x7f781c9adc20>, when = 'setup'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

    @classmethod
    def from_call(cls, func, when, reraise=None):
        #: context of invocation: one of "setup", "call",
        #: "teardown", "memocollect"
        start = time()
        excinfo = None
        try:
>           result = func()

../../../anaconda3/lib/python3.7/site-packages/_pytest/runner.py:229: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../anaconda3/lib/python3.7/site-packages/_pytest/runner.py:201: in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
../../../anaconda3/lib/python3.7/site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
../../../anaconda3/lib/python3.7/site-packages/pluggy/manager.py:92: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../../../anaconda3/lib/python3.7/site-packages/pluggy/manager.py:86: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
../../../anaconda3/lib/python3.7/site-packages/_pytest/runner.py:118: in pytest_runtest_setup
    item.session._setupstate.prepare(item)
../../../anaconda3/lib/python3.7/site-packages/_pytest/runner.py:366: in prepare
    col.setup()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <TestCaseFunction test_author>

    def setup(self):
>       self._testcase = self.parent.obj(self.name)
E       TypeError: Can't instantiate abstract class ApalikeObjectTest with abstract methods test_version

../../../anaconda3/lib/python3.7/site-packages/_pytest/unittest.py:112: TypeError
```

Any idea what may be causing this? I've searched the whole package directory trying to find a place I might have forgotten to declare something, but I got a feeling I might be missing some other silly step.

BTW, I'm using cffconvert version 1.4.0-alpha0 and I've created ApalikeObject.py to support both CFF 1.0.3 and 1.1.0. Should I download CFF 1.0.3 + some unsupported version to check the output?

**Comment by jspaaks** [comment]
Hi Waldir, could you include a pointer to where your code is ? Preferably a permalink if you know how to do that (no biggie if you don't); I can have a look later. Thanks!

**Comment by wleoncio** [comment]
Sure thing, Jurriaan! My working feature branch is https://github.com/wleoncio/cff-converter-python/tree/apaLike_tests, which is a straight-line descendant of the upstream master. The permalink to the latest commit (as of writing) and the problematic file is https://github.com/wleoncio/cff-converter-python/commit/f957f0edcbefc2bfca2f365eb9c9df6c525620d3.

The procedure I used to create that file is the same I did for the 1.1.0 folder:

1. Copy the BibtexObjectTest01.py as ApalikeObjectTest01.py
2. Change `self.bo` to `self.ao` and `BibtexObject` to `ApalikeObject`
3. Use the output of `cffconvert -f apalike -if test/1.0.3/##/CITATION.cff` as a base for adapting the `self.assert*` calls to my case (even though I am running cffconvert > 1.1).

Thank you for taking the time to look into this. After this is worked out I just need to update `cli.py` and issue a Pull Request.

**Comment by jspaaks** [comment]
I think maybe I know what's going wrong (i didnt have time to really look into it yet, so beware) Anyway, the contract says you need a ``test_version`` method on your test class (see [here](https://github.com/wleoncio/cff-converter-python/blob/5c23230e46a998885153643692feda89101fabba/test/contracts/ApalikeObject.py#L22-L24)), but for the 1.0.3 test, that method is missing (see [here](https://github.com/wleoncio/cff-converter-python/blob/f957f0edcbefc2bfca2f365eb9c9df6c525620d3/test/1.0.3/01/ApalikeObjectTest01.py)). 

Hope this helps you, and thanks for taking the time to contribute!

**Comment by wleoncio** [comment]
That was it, thanks!

**Comment by wleoncio** [comment]
Resolved by PR #149.
